jxl/src/api/inner/codestream_parser.rs

```diff
@@ -168,6 +167,18 @@ impl CodestreamParser {
                             break;
                         }
                     }
+                    while self
+                        .sections
+                        .front()
+                        .is_some_and(|s| s.len <= self.ready_section_data)
+                    {
+                        let s = self.sections.pop_front().unwrap();
+                        self.ready_section_data -= s.len;
+                        self.available_sections.push(s);
+                    }
+                    if self.available_sections.is_empty() {
+                        return Err(crate::error::Error::OutOfBounds(0));
```

Comment:
I assume the intention of this change is to return a request for more data to the user if no sections are available.

If so, what I think we should rather be doing is have process_sections return one of:
- Ok(None) ("I processed all the sections I could")
- Ok(Some(usize)) ("I need at least this much data to process the next section")
- Err(some error), mapping Error::OutOfBounds() to Error::SectionTooShort as is done now.

Then, we can discard Ok(None) and map Ok(Some(usize)) into a Error::OutOfBounds()

Reply:
I wanted to do that before calling process sections, but couldn't figure out how to compute the amount of available data - it wasn't ready_sections_data, and ot wasn't sum of section data length. What's the correct calculation?

Reply:
The amount of available section data is ready_sections_data.